src/tcgwars/logic/util/CardList.groovy

```diff
@@ -348,14 +348,14 @@ public class CardList extends ArrayList<Card> {
     }
     if (playerType != TcgStatics.bg().currentThreadPlayerType) {
       TcgStatics.block()
+      def ret = TcgStatics.bg().getClient(playerType).selectCard(new CardSelectUIRequestBuilder()
+        .setMinMax(min, max)
+        .setInfo(info)
+        .setCards(cards)
+        .setCustomCardFilter(filter != null ? filter as CustomCardFilter : null)
+        .setCustomPassFilter(passFilter != null ?  passFilter as CustomPassFilter : null)
+        .setShowAsHidden(hidden))
```

Comment:
I think this should have stayed outside the if clause, don't you think?

Reply:
Honestly didn't look past keeping it as it was, but correcting the casting to the new reference. But yes, I may be wrong but shouldn't this work properly without the ternary if? Will test it out in a bit

Reply:
Yup, seems to be what's causing some Engine errors. Will push a fix on a separate PR in a bit :)

Reply:
Fixed in https://github.com/axpendix/tcgone-engine-contrib/pull/1319